checkov/common/checks/base_check.py

```diff
@@ -18,6 +19,7 @@ def __init__(self, name, id, categories, supported_entities, block_type):
         self.block_type = block_type
         self.supported_entities = supported_entities
         self.logger = logging.getLogger("{}".format(self.__module__))
+        self.evaluated_keys = []
```

Comment:
Could some documentation be added to explain the expectations with this change? Describing the purpose of `evaluated_keys` here or the `get_evaluated_keys` function would be helpful. For one thing: is the syntax assumed to be `dpath` syntax? 
What is the impact if it's omitted (particularly for custom checks). I assume it's just not reported?

Also, documentation on the scan_resource_conf function (or some higher function?) would be helpful to let check implementors know it's something they're expected to update.

Reply:
Hi @robeden , thanks for your comments and sorry for delay!

1. `self.evaluated_keys` should be set for checks that inherit directly from `BaseCheck`, and don't have today their paths of inspected keys, mostly since the checks are more dynamic than the ones that inherit from its child classes.
2. I've added an explanation under the `get_evaluated_keys` and `scan_resource_conf` functions, PTAL and LMK if it's clear enough.
3. If it's omitted the report would get the value of `self.inspected_keys`, so if a previous check in the same run set the value to certain path, and the current check didn't override it, the path would be wrong. That's why I've inserted the `self.evaluated_keys = []` in some of the checks, so if they don't have a relevant key their report doesn't show the previous value.

@nimrodkor Maybe `get_inspected_keys` could use the new `self.evaluated_keys` in `BaseCheck` so its simpler, I'm not sure we need both separated

Reply:
Documentation looks good. Thanks.